src/settings/channel-range-advanced.js

```diff
@@ -0,0 +1,260 @@
+//
+// Copyright (C) 2017 University of Dundee & Open Microscopy Environment.
+// All rights reserved.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+import Context from '../app/context';
+import Misc from '../utils/misc';
+import {Converters} from '../utils/converters';
+import {inject, customElement, bindable, BindingEngine} from 'aurelia-framework';
+import {spinner} from 'jquery-ui/ui/widgets/spinner';
+import {slider} from 'jquery-ui/ui/widgets/slider';
+
+/**
+ * Extends the channel range (showing advanced rendering settings)
+ */
+@customElement('channel-range-advanced')
+@inject(Context, Element, BindingEngine)
+export default class ChannelRangeAdvanced  {
+    /**
+     * channel informetion (bound in template)
+     * @memberof ChannelRangeAdvanced
+     * @type {Object}
+     */
+    @bindable channel = null;
+
+    /**
+     * the channel index
+     * @memberof ChannelRangeAdvanced
+     * @type {number}
+     */
+    @bindable index = null;
+
+    /**
+     * the mode
+     * @memberof ChannelRangeAdvanced
+     * @type {number}
+     */
+    @bindable mode = null;
+
+    /**
+     * the revision count (used for history)
+     * @memberof ChannelRangeAdvanced
+     * @type {number}
+     */
+    @bindable revision = 0;
+    revisionChanged(newVal, oldVal) {
+        let imgInf =
+            this.context.getSelectedImageConfig().image_info;
+        imgInf.initial_values = true;
+        this.updateUI();
+    }
+
+    /**
+     * the slider min/max for gamma values
+     * @memberof ChannelRangeAdvanced
+     * @type {Array.<number>}
+     */
+    GAMMA_MIN_MAX = [0,4];
+
+    /**
+     * the families allowed
+     * @memberof ChannelRangeAdvanced
+     * @type {Array.<string>}
+     */
+    families = [
+        'linear',
```

Comment:
ideally that should be retrieved from the server (enum)
This could be improved later one
add a TODO for now